test/e2e/scenario/recorder.scenario.ts

```diff
@@ -654,17 +694,305 @@ describe('recorder', () => {
+        await sleep(210)
+        const nextVisualViewportDimension = await getVisualViewport()
+
+        await flushEvents()
+        const segment = getLastSegment(events)
+        const visualViewportRecords = findAllVisualViewport(segment)
+        const lastVisualViewportRecord = visualViewportRecords.slice(-1)[0]
+
+        // SDK returns Visual Viewport object
+        expectToBeNearby(lastVisualViewportRecord.data.scale, nextVisualViewportDimension.scale)
+        expectToBeNearby(lastVisualViewportRecord.data.width, nextVisualViewportDimension.width)
+        expectToBeNearby(lastVisualViewportRecord.data.height, nextVisualViewportDimension.height)
+        expectToBeNearby(lastVisualViewportRecord.data.offsetLeft, nextVisualViewportDimension.offsetLeft)
+        expectToBeNearby(lastVisualViewportRecord.data.offsetTop, nextVisualViewportDimension.offsetTop)
+        expectToBeNearby(lastVisualViewportRecord.data.pageLeft, nextVisualViewportDimension.pageLeft)
+        expectToBeNearby(lastVisualViewportRecord.data.pageTop, nextVisualViewportDimension.pageTop)
+
+        // With correct transformation
+        const finalScaleAmount = nextVisualViewportDimension.scale
+        expectToBeNearby(lastVisualViewportRecord.data.scale, finalScaleAmount)
+        expectToBeNearby(lastVisualViewportRecord.data.width, initialVisualViewportDimension.width / finalScaleAmount)
+        expectToBeNearby(lastVisualViewportRecord.data.height, initialVisualViewportDimension.height / finalScaleAmount)
+
+        expect(lastVisualViewportRecord.data.offsetLeft).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.offsetTop).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.pageLeft).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.pageTop).toBeGreaterThan(0)
+      })
+  })
 })
 
-function getFirstSegment(events: EventRegistry) {
+export function getFirstSegment(events: EventRegistry) {
   return events.sessionReplay[0].segment.data
 }
 
-function getLastSegment(events: EventRegistry) {
+export function getLastSegment(events: EventRegistry) {
   return events.sessionReplay[events.sessionReplay.length - 1].segment.data
 }
 
-function initRumAndStartRecording(initConfiguration: RumInitConfiguration) {
+export function initRumAndStartRecording(initConfiguration: RumInitConfiguration) {
   window.DD_RUM!.init(initConfiguration)
   window.DD_RUM!.startSessionReplayRecording()
 }
+
+export async function pinchZoom(xChange = 50, durationMS = 400) {
+  const xBase = 180
+  const yBase = 180
+  const xOffsetFingerTwo = 25
+  const actions = [
+    {
+      type: 'pointer',
+      id: 'finger1',
+      parameters: { pointerType: 'touch' },
+      actions: [
+        { type: 'pointerMove', duration: 0, x: xBase, y: yBase },
+        { type: 'pointerDown', button: 0 },
+        { type: 'pause', duration: 150 },
+        { type: 'pointerMove', duration: durationMS, origin: 'pointer', x: -xChange, y: 0 },
```

Comment:
it could be interesting to add some explanation for all the magic numbers, are they random or are they important?